bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -229,18 +243,51 @@ impl Client {
             user_id,
             device_id,
             homeserver_url: _,
+            oidc_data,
             sliding_sync_proxy,
         } = session;
 
-        let session = matrix_sdk::matrix_auth::Session {
-            meta: matrix_sdk::SessionMeta {
-                user_id: user_id.try_into()?,
-                device_id: device_id.into(),
-            },
-            tokens: matrix_sdk::matrix_auth::SessionTokens { access_token, refresh_token },
-        };
+        if let Some(oidc_data) = oidc_data {
+            // Restore using an OIDC FullSession.
+            let oidc_data = serde_json::from_str::<OidcUnvalidatedSessionData>(&oidc_data)?
+                .validate()
+                .context("OIDC metadata validation failed.")?;
+            let latest_id_token = oidc_data.latest_id_token.and_then(|s| s.try_into().ok());
```

Comment:
This would error out if `latest_id_token` conversion failed, is that not preferable over silently using `None` instead if conversion fails?

```suggestion
            let latest_id_token = oidc_data.latest_id_token.and_then(|s| s.try_into()).transpose()?;
```